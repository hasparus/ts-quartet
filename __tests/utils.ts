import { getDescription } from "./getDescription";

export function snapshot(validator: any) {
  const description = getDescription(validator);

  expect(description).toMatchSnapshot();
}
export function getExplanation(
  validator: { (arg0: any): void; explanations: any },
  value: any
) {
  validator(value);
  return validator.explanations;
}
export function isValid(validator: (arg0: any) => any, value: any) {
  const valid = validator(value);
  if (!valid) {
    console.log(JSON.stringify(value, null, 2));
  }
  expect(valid).toBe(true);
}
export function isNotValid(validator: (arg0: any) => any, value: any) {
  const valid = validator(value);
  if (valid) {
    console.log(JSON.stringify(value, null, 2));
  }
  expect(valid).toBe(false);
}

export function tables(validator: any, valids: any[], invalids: any[]) {
  for (const valid of valids) {
    isValid(validator, valid);
  }
  for (const invalid of invalids) {
    isNotValid(validator, invalid);
  }
}
