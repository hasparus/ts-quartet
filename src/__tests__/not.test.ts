import { v } from "../index";
import { tables, snapshot } from "./utils";

describe("not", () => {
  test("constant", () => {
    const checkNot42 = v(v.not(42));
    tables(checkNot42, [41, "42", 43, null, false], [42]);
    snapshot(checkNot42);
  });
});
