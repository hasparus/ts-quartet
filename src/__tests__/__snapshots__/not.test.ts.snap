// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not constant 1`] = `
Object {
  "_": "function validator(value) {
    validator.explanations = []
    return !validator['not-0'](value)
  }",
  "explanations": Array [],
  "not-0": "function (value) { return value === c; }",
  "not-0.explanations": Array [],
}
`;
